src/auth/mod.ts

```diff
@@ -15,33 +14,131 @@ interface DecodedToken {
 
 const logger = getLogger(import.meta);
 
-function isSubInPodsConfig(payload: DecodedToken): boolean {
-  return podsConfig.pods.some((pod) =>
-    pod.sub === payload.sub &&
-    pod.serviceaccount.uid === payload["kubernetes.io"]?.serviceaccount.uid
+export async function authenicateRequestAndReturnPodServiceAccount(
+  token: string,
+): Promise<string> {
+  const payload = await verifyToken(token);
+
+  if (!payload["kubernetes.io"]) {
```

Comment:
what about a potential case where the serviceaccount is null? hint: let's not add a check for all cases, rather check that the value is indeed existing at the end

Reply:
The plan is to reject a request if the payload doesn't have a service account. I didn't understand by what you meant "let's not add a check for all cases"